itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/txn/compactor/TestCrudCompactorOnTez.java

```diff
@@ -87,6 +90,262 @@
 @SuppressWarnings("deprecation")
 public class TestCrudCompactorOnTez extends CompactorOnTezTest {
 
+
+  @Test
+  public void testRebalanceCompactionWithParallelDelete() throws Exception {
+    conf.setBoolVar(HiveConf.ConfVars.COMPACTOR_CRUD_QUERY_BASED, true);
+    conf.setBoolVar(HiveConf.ConfVars.HIVE_COMPACTOR_GATHER_STATS, false);
+    conf.setBoolVar(HiveConf.ConfVars.HIVESTATSAUTOGATHER, false);
+
```

Comment:
please create a parametrized test and validate for `TXN_WRITE_X_LOCK=false` as well

Reply:
that flag controls optimistic/pessimistic concurrency control mode